### PR TITLE
feat: add route for School of Economics & Management, Tongji University (同济大学经济与管理学院)

### PR DESCRIPTION
**Pull request by sitdownkevin**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/tongji/sem
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [X] New Route / 新的路由
  - [X] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [X] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [X] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/tongji/sem/notice.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '经济与管理学院通知',
+    maintainers: ['sitdownkevin'],
+    url: 'https://sem.tongji.edu.cn/semch/category/frontpage/notice',
```
Comment (TonyRL) [inline lib/routes/tongji/sem/notice.ts:19]:
```suggestion
    url: 'sem.tongji.edu.cn/semch/category/frontpage/notice',
```

File: lib/routes/tongji/sem/_utils.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+export async function getNotifByPage(pageNumber: number) {
+    const pageUrl: string = `https://sem.tongji.edu.cn/semch/category/frontpage/notice/page/${pageNumber}`;
+
+    try {
+        const response = await got.get(pageUrl, {
+            headers: {
+                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/38.0.101.76 Safari/537.36',
```
Comment (TonyRL) [inline lib/routes/tongji/sem/_utils.ts:11]: RSSHub will use a randomised user agent of [Chrome on mac](https://github.com/DIYgod/RSSHub/blob/263d6b818ea551e9a498207ba1f0a912b6370224/lib/config.ts#L344) by default. Does the site only works with this fixed old version of Chrome on Windows?

File: lib/routes/tongji/sem/_utils.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        // console.error(error);
+    }
+    return [];
+}
+
+export async function getLastPageNumber() {
+    try {
+        const response = await got.get('https://sem.tongji.edu.cn/semch/category/frontpage/notice', {
+            headers: {
+                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/38.0.101.76 Safari/537.36',
```
Comment (TonyRL) [inline lib/routes/tongji/sem/_utils.ts:44]: RSSHub will use a randomised user agent of [Chrome on mac](https://github.com/DIYgod/RSSHub/blob/263d6b818ea551e9a498207ba1f0a912b6370224/lib/config.ts#L344) by default. Does the site only works with this fixed old version of Chrome on Windows?

File: lib/routes/tongji/sem/notice.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    description: ``,
+};
+
+import { getNotifByPage } from './_utils';
+
+async function handler() {
+    const promises = [];
+    for (let i = 1; i <= 20; i++) {
+        promises.push(getNotifByPage(i));
+    }
```
[the remark this answers is not in the document]
Comment (sitdownkevin) [reply lib/routes/tongji/sem/notice.ts:30]: The maximum number of pages was too large, so I only retrieved data from the first 20 pages.

[the remark this answers is not in the document]
Comment (TonyRL) [reply lib/routes/tongji/sem/notice.ts:30]: That's why I said fetch data from the **first page only**.

File: lib/routes/tongji/sem/_utils.ts (hunk 3 above)
Comment (sitdownkevin) [reply lib/routes/tongji/sem/_utils.ts]: I've fixed it.

File: lib/routes/tongji/sem/notice.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (sitdownkevin) [reply lib/routes/tongji/sem/notice.ts]: I've fixed it.

File: lib/routes/tongji/sem/_utils.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            } else {
+                // console.error('Failed to extract last page number.');
+            }
+        }
+    } catch {
+        // console.error(error);
+    }
+
+    return -1;
+}
```
Comment (TonyRL) [inline lib/routes/tongji/sem/_utils.ts:71]: Remove unused function

```suggestion
```

File: lib/routes/tongji/sem/notice.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+// Warning: The author still knows nothing about javascript!
+import { Route } from '@/types';
+
```
Comment (TonyRL) [inline lib/routes/tongji/sem/notice.ts:3]:
```suggestion
import { Route } from '@/types';
import { getNotifByPage } from './_utils';

```

File: lib/routes/tongji/sem/notice.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    name: '经济与管理学院通知',
+    maintainers: ['sitdownkevin'],
+    url: 'sem.tongji.edu.cn/semch/category/frontpage/notice',
+    handler,
+    description: ``,
+};
+
+import { getNotifByPage } from './_utils';
+
+async function handler() {
```
Comment (TonyRL) [inline lib/routes/tongji/sem/notice.ts:26]:
```suggestion
};

async function handler() {
```

File: lib/routes/tongji/sem/_utils.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import { load } from 'cheerio';
+import { parseDate } from '@/utils/parse-date';
+
+import { config } from '@/config';
+
+export async function getNotifByPage(pageNumber: number) {
+    const pageUrl: string = `https://sem.tongji.edu.cn/semch/category/frontpage/notice/page/${pageNumber}`;
+
+    try {
+        const response = await got.get(pageUrl, {
```
Comment (TonyRL) [inline lib/routes/tongji/sem/_utils.ts:11]: Use the [redirected URL](https://github.com/DIYgod/RSSHub/actions/runs/11757146978/job/32754065160#step:13:21) instead.

**Pull request merged**